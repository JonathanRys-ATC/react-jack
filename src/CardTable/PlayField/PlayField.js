import React, { Component } from 'react'

import Hand from './Hand/Hand'

export default class PlayField extends Component {
    render() {
        return (
            <div>
                <Hand faceUp={true} />
                PlayField should consist of two hands, dealer and player
                {/* <PlayerHand /> */}
            </div>
        )
    }
}