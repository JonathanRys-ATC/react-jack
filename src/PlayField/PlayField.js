import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Dealer from './Dealer/Dealer'
import Deck from './Deck/Deck'
import Player from './Player/Player'

import { connect } from 'react-redux'

const mapStateToProps = (state) => {
    return {
        hands: state.player.hands
    }
}

export class PlayField extends Component {
    render() {
        return (
            <div>
                <Dealer />
                <Deck />
                {this.props.hands.map((hand, index) => <Player key={`hand-${index}`} hand={hand} faceUp={index} />)}
            </div>
        )
    }
}

PlayField.propTypes = {
    hands: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(PlayField)
