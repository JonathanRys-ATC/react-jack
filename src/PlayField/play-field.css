.PlayField_main {
    border: 3px groove #aaa;
    border-radius: 10px;
    margin-top: 20px;
    padding-top: 10px;
}