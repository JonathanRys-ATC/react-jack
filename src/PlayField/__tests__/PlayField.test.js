import React from 'react'
import { shallow } from 'enzyme'

import PlayField from '../PlayField'

import mockStore from "../../mockStore"

const mockProps = {
    ...mockStore,
    giveCard: jest.fn(),
    giveDealerCard: jest.fn(),
    clearCard: jest.fn()
}

describe("Test PlayField component", () => {
    it("matches the snapshot", () => {
        const testProps = { ...mockProps }
        const component = shallow(<PlayField {...testProps} />)
        expect(component).toMatchSnapshot()
    })
})