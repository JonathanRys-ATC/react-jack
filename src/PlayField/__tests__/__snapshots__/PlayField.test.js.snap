// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlayField component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayField
    clearCard={[MockFunction]}
    dealer={
      Object {
        "avatar": "./images/dealer.png",
        "busted": Array [
          false,
        ],
        "hand": Array [
          "HA",
          "C5",
        ],
        "handFlipped": false,
        "hasBlackjack": Array [
          false,
        ],
        "name": "Dealer",
        "score": Array [
          0,
        ],
      }
    }
    deck={
      Object {
        "deck": Array [],
        "drawnCard": null,
      }
    }
    giveCard={[MockFunction]}
    player={
      Object {
        "avatar": "./images/default.png",
        "balance": 0,
        "busted": Array [
          Array [
            false,
          ],
        ],
        "currentBet": 0,
        "hands": Array [
          Array [
            "SA",
            "CK",
          ],
          Array [
            "D2",
            "HJ",
          ],
        ],
        "hasBlackjack": Array [
          Array [
            false,
          ],
        ],
        "hasInsurance": Array [
          Array [
            false,
          ],
        ],
        "name": "Player1",
        "playerIndex": 1,
        "score": Array [
          Array [
            0,
          ],
        ],
        "splitHand": false,
      }
    }
    turn={
      Object {
        "errMsg": "",
        "error": false,
        "isPlaying": true,
        "numPlayers": 2,
        "playersTurn": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Row
          tag="div"
        >
          <Dealer
            clearCard={[MockFunction]}
            drawnCard={null}
            hand={
              Array [
                "HA",
                "C5",
              ]
            }
            playersTurn={0}
            takeCard={undefined}
          />
        </Row>,
        <Row
          tag="div"
        >
          <Deck />
           
        </Row>,
        <Row
          tag="div"
        >
          <Player
            clearCard={[MockFunction]}
            drawnCard={null}
            hands={
              Array [
                Array [
                  "SA",
                  "CK",
                ],
                Array [
                  "D2",
                  "HJ",
                ],
              ]
            }
            playersTurn={0}
            takeCard={[MockFunction]}
          />
        </Row>,
      ],
      "className": "PlayField_main",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Dealer
            clearCard={[MockFunction]}
            drawnCard={null}
            hand={
              Array [
                "HA",
                "C5",
              ]
            }
            playersTurn={0}
            takeCard={undefined}
          />,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "clearCard": [MockFunction],
            "drawnCard": null,
            "hand": Array [
              "HA",
              "C5",
            ],
            "playersTurn": 0,
            "takeCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Deck />,
            " ",
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Player
            clearCard={[MockFunction]}
            drawnCard={null}
            hands={
              Array [
                Array [
                  "SA",
                  "CK",
                ],
                Array [
                  "D2",
                  "HJ",
                ],
              ]
            }
            playersTurn={0}
            takeCard={[MockFunction]}
          />,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "clearCard": [MockFunction],
            "drawnCard": null,
            "hands": Array [
              Array [
                "SA",
                "CK",
              ],
              Array [
                "D2",
                "HJ",
              ],
            ],
            "playersTurn": 0,
            "takeCard": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Row
            tag="div"
          >
            <Dealer
              clearCard={[MockFunction]}
              drawnCard={null}
              hand={
                Array [
                  "HA",
                  "C5",
                ]
              }
              playersTurn={0}
              takeCard={undefined}
            />
          </Row>,
          <Row
            tag="div"
          >
            <Deck />
             
          </Row>,
          <Row
            tag="div"
          >
            <Player
              clearCard={[MockFunction]}
              drawnCard={null}
              hands={
                Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ]
              }
              playersTurn={0}
              takeCard={[MockFunction]}
            />
          </Row>,
        ],
        "className": "PlayField_main",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Dealer
              clearCard={[MockFunction]}
              drawnCard={null}
              hand={
                Array [
                  "HA",
                  "C5",
                ]
              }
              playersTurn={0}
              takeCard={undefined}
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clearCard": [MockFunction],
              "drawnCard": null,
              "hand": Array [
                "HA",
                "C5",
              ],
              "playersTurn": 0,
              "takeCard": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Deck />,
              " ",
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Player
              clearCard={[MockFunction]}
              drawnCard={null}
              hands={
                Array [
                  Array [
                    "SA",
                    "CK",
                  ],
                  Array [
                    "D2",
                    "HJ",
                  ],
                ]
              }
              playersTurn={0}
              takeCard={[MockFunction]}
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "clearCard": [MockFunction],
              "drawnCard": null,
              "hands": Array [
                Array [
                  "SA",
                  "CK",
                ],
                Array [
                  "D2",
                  "HJ",
                ],
              ],
              "playersTurn": 0,
              "takeCard": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
